commons/src/main/java/org/sonarsource/analyzer/commons/ProgressReport.java

```diff
@@ -54,22 +55,33 @@ public ProgressReport(String threadName, long period) {
 
   @Override
   public void run() {
+    log(count + " source " + pluralizeFile(count) + " to be " + adjective);
     while (!Thread.interrupted()) {
       try {
         Thread.sleep(period);
-        synchronized (this) {
-          log(currentFileNumber + "/" + count + " files " + adjective + ", current file: " + currentFilename);
-        }
+        log(currentFileNumber + "/" + count + " " + pluralizeFile(currentFileNumber) + " " + adjective + ", current file: " + currentFilename);
       } catch (InterruptedException e) {
-        Thread.currentThread().interrupt();
+        thread.interrupt();
         break;
       }
     }
-    synchronized (this) {
-      if (success) {
-        log(count + "/" + count + " source files have been " + adjective);
-      }
+    if (success) {
+      log(count + "/" + count + " source " + pluralizeFile(count) + " " + pluralizeHas(count) + " been " + adjective);
+    }
+  }
+
+  private static String pluralizeFile(long count) {
+    if (count == 1) {
```

Comment:
Nitpick (and same in `pluralizeHas`):
```suggestion
    if (count == 1L) {
```